### PR TITLE
FEWD-9745: 3.1.4 relnotes

### PR DESCRIPTION
**Pull request by ElliotFrancisHunter**
Added release notes for 3.1.4 maintenance release.

Also tidied up older release notes docs and ensured the format was consistent with dev release notes page and between the docs themselves.

### CODE REVIEW

File: modules/ROOT/pages/_partials/release_notes/sync-gateway-3-0-4-release-note.adoc [hunk 1]
```diff
@@ -1,51 +1,34 @@
 == 3.0.4 -- October 2022
 
 [#maint-3-0-4]
-=== Enhancements
```
Comment (osfameron) [inline modules/ROOT/pages/_partials/release_notes/sync-gateway-3-0-4-release-note.adoc:4]: Confusing that you're editing a 3.0 relnote (unused) in this commit

File: modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-0-release-note.adoc [hunk 2]
```diff
@@ -8,6 +8,42 @@ include::ROOT:whatsnew.adoc[tag=scopes-and-collections]
 
 
 [#maint-3-1-0]
```
Comment (osfameron) [inline modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-0-release-note.adoc:10]: This is just an unrelated re-ordering of the earlier 3.1.x relnotes?

**Review by osfameron — APPROVED** [review]
👍

File: modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-0-release-note.adoc (hunk 2 above)
Comment (ElliotFrancisHunter) [reply modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-0-release-note.adoc:10]: Yup

**Pull request merged**